Onprem 326/machine runner preview migration (#8359)
circleci/circleci-docs: jekyll/_cci2/migrating-from-launch-agent-to-the-machine-runner-agent-3-0-preview.adoc

---
contentTags: 
  platform:
  - Cloud
---

>  = Migrating from launch agent to the machine runner 3.0 preview

Reviewer: ```suggestion
= Migrate from launch agent to machine runner 3.0 - Open preview
```

Reviewer: Could you also change the filename to match but without `- Open preview` ?

:page-layout: classic-docs
:page-liquid:
:page-description: Steps to migrate from using launch agent to the machine runner 3.0 preview
:icons: font
:toc: macro
:toc-title:
:machine:
:experimental:
:linux:

This page describes how to migrate an existing launch agent installation to machine runner 3.0 preview. The machine runner 3.0 is a new way to run machine jobs for self-hosted runners, with improved network resiliency and simpler installation management strategy utilizing operating system packages.

> NOTE: Machine runner 3.0 is currently in open preview

Reviewer: ```suggestion
NOTE: Machine runner 3.0 is currently in **open preview**, and is currently available on **Linux** only.
```

> NOTE: Machine runner 3.0 is currently only available on Linux

Reviewer: ```suggestion
```

> Migrating from launch agent to the machine runner 3.0 preview is a straightforward process. The prerequisites remain the same. First uninstall and remove launch agent, then, install the machine runner 3.0 agent. The configuration file is 1:1 compatible between the agents so no modifications are needed during the migration

Reviewer: ```suggestion
Migrating from launch agent to machine runner 3.0 is a straightforward process. The prerequisites remain the same. First, uninstall and remove launch agent, then, install the machine runner 3.0 agent. The configuration file is 1:1 compatible between the agents so no modifications are needed during the migration
```

[#uninstall-launch-agent]
== 1. Uninstall launch agent

The first step is to uninstall launch agent. The exact process depends on the platform, as described in the following sections.

> If the agent has been installed as a `systemd` service, run the following commands to stop and uninstall the service unit, and delete the service file (the new agent will install a new service unit):

Reviewer: ```suggestion
. If the agent has been installed as a `systemd` service, run the following commands to stop and uninstall the service unit, and delete the service file (the new agent will install a new service unit):
```

sudo systemctl stop circleci.service
sudo systemctl disable circleci.service

sudo rm /usr/lib/systemd/system/circleci.service
```

. The launch agent binary can now be removed from the machine. The default install location is `/opt/circleci/circleci-launch-agent`, but this may differ on your machine if the agent was installed somewhere else. The entire `/opt/circleci` subdirectory can be removed at this point.

. Finally, move the agent configuration file. The default location for this is `/etc/opt/circleci/launch-agent-config.yaml`. Move the file to `/etc/circleci-runner/circleci-runner-config.yaml` using the following command:
+
```shell
mv /etc/opt/circleci/launch-agent-config.yaml /etc/circleci-runner/circleci-runner-config.yaml
```
+
Alternatively, you can let the machine agent install process create a config file template and copy the existing config into the newly created file (CROSSLINK)

[#uninstalling-launch-agent-linux-se]
==== RHEL/Rocky Linux

In addition to the steps listed above, the SELinux policy should be uninstalled using the following command:

```shell
sudo semodule -r circleci_launch_agent
```

[#install-machine-agent]
== 2. Installing the Machine Agent

> Packaged versions of the machine runner 3.0 preview are available for Linux for a streamlined installation experience.

Reviewer: ```suggestion
Packaged versions of the machine runner 3.0 are available for Linux for a streamlined installation experience.
```

. At present deb and rpm packages are available for linux. Use the following commands to install them
+
```shell
# DEB Commands
# This script installs the package repository to the package manager
curl -s https://packagecloud.io/install/repositories/circleci/runner/script.deb.sh | sudo bash
sudo apt install circleci-runner
```
+
```shell
# RPM Commands
# This script installs the package repository to the package manager
curl -s https://packagecloud.io/install/repositories/circleci/runner/script.rpm.sh | sudo bash
sudo yum install circleci-runner
```
+

> The packages installs the latest version of the agent. They also create a circleci user as well as required directories for operation of the agent.

Reviewer: ```suggestion
The packages install the latest version of the agent. They also create a circleci user as well as required directories for operation of the agent.
```

> . The packages creates a template configuration file at `/etc/circleci-runner/circleci-runner-config.yaml`. This configuration is 1:1 compatible with the launch agent configuration, apart from the auto-update feature. Since auto-update has been removed from the new agent, you should use the appropriate package management update tools to get new versions of the package when they are released. When using the `apt` package to update the agent package, you will need to clear the configuration file modified prompt. This can be avoided using the `--force-confold` flag or by removing the modified configuration file before upgrading. 

Reviewer: ```suggestion
. The packages create a template configuration file at `/etc/circleci-runner/circleci-runner-config.yaml`. This configuration is 1:1 compatible with the launch agent configuration, apart from the auto-update feature. Since auto-update has been removed from the new agent, you should use the appropriate package management update tools to get new versions of the package when they are released. When using the `apt` package to update the agent package, you will need to clear the configuration file modified prompt. This can be avoided using the `--force-confold` flag or by removing the modified configuration file before upgrading. 
```

[#uninstall-launch-agent]
== 1. Uninstalling Launch Agent

The first step is to uninstall launch agent. The exact process depends on the platform, as described in the following sections.

If the agent has been installed as a `systemd` service, run the following commands the stop and uninstall the service unit, and delete the service file. (The new agent will install a new service unit):

+
```shell
sudo systemctl stop circleci.service
sudo systemctl disable circleci.service

sudo rm /usr/lib/systemd/system/circleci.service
```

. The launch agent binary can now be removed from the machine. The default install location is `/opt/circleci/circleci-launch-agent`, but this may differ on your machine if the agent was installed somewhere else. The entire `/opt/circleci` subdirectory can be removed at this point.

. Finally, move the agent configuration file. The default location for this is `/etc/opt/circleci/launch-agent-config.yaml`. Move the file to `/etc/circleci-runner/circleci-runner-config.yaml` using the following command:

+
```shell

> mv /etc/opt/circleci/launch-agent-config.yaml /etc/circleci-runner/c sircleci-runner-config.yaml

Reviewer 2: ```suggestion
mv /etc/opt/circleci/launch-agent-config.yaml /etc/circleci-runner/circleci-runner-config.yaml
```
